feat: wire application constraints in the apiserver

nvinuesa opened the pull request:
This patch wires the recently added application constraints methods from the application domain into the apiserver.

As a side quest, the `Constraints()` method from the legacy state `Application` struct was made private, since it's only used inside the legacy `state/application.go` and making it private we ensure it's not used elsewhere. It must be noted that we cannot remove it because setting the constraints in mongo has to continue until the machine domain is finished.


## QA steps

This changes should only impact application constraints, so setting and retrieving should be enough.
First bootstrap and add a model, then:

```
$ juju deploy ubuntu
$ juju constraints ubuntu

$ juju set-constraints ubuntu mem=42M
$ juju constraints ubuntu
mem=42M
```
Scaling the application should continue to work (at least with the useless constraint we just set):
```
$ juju add-unit ubuntu
```

## Links


**Jira card:** JUJU-7502

jack-w-shaw reviewed (commented):
Do the `apiserver/facade/client/application` changes really require no test changes?!?

SimonRichardson commented on apiserver/facades/client/application/application_get.go:
```diff
@@ -73,9 +73,17 @@ func (api *APIBase) getConfig(
+		} else if err != nil {
+			return params.ApplicationGetResults{}, errors.Trace(err)
+		}
+		cons, err = api.applicationService.GetApplicationConstraints(ctx, appID)
```
This should have been just the name, not the application ID, similar to charm, use the natural key.

nvinuesa replied:
` GetApplicationConfig()`, `GetApplicationTrustSetting()`, ` UnsetApplicationConfigKeys()`, `SetApplicationConfig()` all these methods use app ID as ident. There is some homogenization work on the app domain but it should not be part of this patch.

SimonRichardson reviewed (commented):
Add tests to `apiserver/facades/client/application`

nvinuesa commented:
@jack-w-shaw 
> Do the apiserver/facade/client/application changes really require no test changes?!?

It turns out we had removed a bunch of tests from the apiserver before starting this epic, there was only one for constraints. I added new tests for setting/getting them and using mocks.

nvinuesa commented:
/merge
